src/effects/EffectUI.cpp

```diff
@@ -357,73 +344,46 @@ wxPanel *EffectUIHost::BuildButtonBar(wxWindow *parent, bool graphicalUI)
    {
       S.SetBorder( margin );
 
-      if (!mIsGUI)
-      {
-         mMenuBtn = S.Id( kMenuID )
-            .ToolTip(XO("Manage presets and options"))
-            .AddButton( XXO("&Manage"), wxALIGN_CENTER | wxTOP | wxBOTTOM );
-      }
-      else
+      if (!mwState.expired())
       {
-         mMenuBtn = S.Id( kMenuID )
-            .ToolTip(XO("Manage presets and options"))
-            .Name(XO("&Manage"))
-            .AddBitmapButton( CreateBitmap(effect_menu_xpm, true, true) );
-         mMenuBtn->SetBitmapPressed(CreateBitmap(effect_menu_xpm, false, true));
+         mEnableBtn = S.Id(kEnableID)
+            .Position(wxALIGN_CENTER | wxTOP | wxBOTTOM)
+            .Name(XO("Enable"))
+            .AddBitmapButton(mRealtimeEnabledBM);
+
+         mEnableBtn->SetBitmapDisabled(mRealtimeDisabledBM);
+
+         S.AddSpace( 5, 5 );
       }
 
-      S.AddSpace( 5, 5 );
+      mMenuBtn = S.Id( kMenuID )
+         .ToolTip(XO("Manage presets and options"))
+         .AddButton( XXO("Presets && settings"), wxALIGN_CENTER | wxTOP | wxBOTTOM );
 
       if (!mIsBatch)
       {
-         if (!mIsGUI)
+         if (mSupportsRealtime)
          {
-            if (mSupportsRealtime)
-            {
-               mPlayToggleBtn = S.Id( kPlayID )
-                  .ToolTip(XO("Start and stop playback"))
-                  .AddButton( XXO("Start &Playback"),
-                              wxALIGN_CENTER | wxTOP | wxBOTTOM );
-            }
-            else if (
-               (mEffectUIHost.GetDefinition().GetType() != EffectTypeAnalyze) &&
-               (mEffectUIHost.GetDefinition().GetType() != EffectTypeTool) )
+            if (mwState.expired())
             {
-               mPlayToggleBtn = S.Id( kPlayID )
-                  .ToolTip(XO("Preview effect"))
-                  .AddButton( XXO("&Preview"),
+               S.AddSpace( 5, 5 );
+
+               mPlayToggleBtn = S.Id(kPlayID)
+                  .ToolTip(XO("Start and stop preview"))
+                  .AddButton( XXO("Start &Preview"),
```

Comment:
I have checked all the design sources - button is always called "Preview". Also, it should be "grouped" with the "Apply", right now it leans to the left (i. e. it is near the "Presets" button)

Reply:
Agree on calling it "preview" (changing to "stop preview" after it's been pressed). Regarding the grouping, it apparently was difficult and martin didn't mind it ungrouped iirc.

Reply:
Well, I know how to do such grouping :-)